src/compiler/types.ts

```diff
@@ -4061,6 +4062,8 @@ namespace ts {
         length: number | undefined;
         messageText: string | DiagnosticMessageChain;
         category: DiagnosticCategory;
+        /** May store more in future. For now, this will simply be `true` to indicate when a diagnostic is an unused-identifier diagnostic. */
+        reportsUnused?: {};

```

Comment:
nit. why not call both this property and the one on `DiagnosticMessage` `isUnused` or `isUnusedDeclaration`

@amcasey what does Roslyn call this property?

Reply:
I think you're asking about `WellKnownDiagnosticTags.Unnecessary`

Reply:
a bit weird ... but sure.